### ISSUE TITLE
File Browser enhancements

### ISSUE THREAD

**Issue by worldexplorer** [koreader/koreader#3756]
* KOReader version:  FEATURE__REQUEST
* Device: kobo glo hd

#### Issue

glitches & features of the main browser
~~~~~~~~~~~~~~~~~~~~~~~~~
1) if I CUT a file , then I open an EPUB, return to the file browser, I can't PASTE because what i COPIED is lost. I need to go find it again, change the folder again and PASTE;

2) in the file browser, long click on the empty folder doesn't bring any context menu. I can't PASTE into the folder that has no files;

3) PASTing into ".." pastes into the parent folder; it's logical but not the expected outcome;

4) context menu, besides "Set as Home Directory", may have "Save EPUBs downloaded from wiki into THIS folder"

5) I'm missing multiple files selection, probably with checkboxes and changing the background color; for further COPY / CUT + PASTE with a review of which files are selected - as a plain list (I may have selected EPUBs from multiple folders) of EPUBs + respective *.sdr

6) please add "Move To" menu item for selected files, with a folder chooser as a new pop-up, instead of walking away and PASTE into another folder - sample of UI is in AntTek explorer android app; for portable touch devices, one instance of the interface is preferred instead of multitasking Windows-style "open new window", norton commander - like panels, and implicitly changing directories like now in koreader

I'll pay $20 ; I just need these features :)

**Comment by Frenzie** [comment]
Pinging @robert00s who's been adding some good stuff to the file browser.

> PASTing into ".." pastes into the parent folder; it's logical but not the expected outcome; files pasted just disappear from the view (being one level up) which is unexpected; the popup with "PASTE" should probably NOT show up on ".." element, but SHOULD on the empty field if the folder is empty, or there are 4 files and the rest of the screen is empty (must be long-clickable, I think)

Don't have the program at hand atm but wasn't this added to the + menu as a workaround? :-) (Set download folder could also go in there.)

**Comment by poire-z** [comment]
I consider 3. , as you describe how it is currently, as a big feature and time saver (that answers your 6. ):
when you are in a dir, and want to move a file up a directory or into a subdir, you just hold on it, cut, hold on ".." or on the subdir, and paste. Done.
On slow eink screen, I don't want to cut, tap on the directory I want to go in, display the whole new directory, find an empty space to hold on to to be able to paste.
(I'm not even sure it was intended to be such a good feature, may be it's just a side effect of pasting expecting a directory and being given a directory).

About 4. (and an other comment in another issue of yours), I reckon the `Wikipedia settings/Set Wikipedia 'Save as EPUB' directory` is a bit rudimentory, and could probably be changed to a FileChooser widget.
I guess that when you are in a wikipedia epub saving campaign, you are on a same topic, so you don't need to be asked every time where to save it. You'd set it at start with a hopefully better/easier widget, and use that one for some time, till you get to another topic.
I don't think it should be in the hold menu (because if so, why not all the others: OPDS, cloud storage, the various plugins that download stuff...?)

But why not: have a `Wikipedia/` directory, with subdirs `Wikipedia/Science/` `Wikipedia/Litterature/` ... , having the EPUB saved in `Wikipedia/` , and (with what I described in my first point above) moving them later in the adequate subdirs should take less steps than what you would get with your other requests answered, I think.

**Comment by Frenzie** [comment]
Agreed on 3. I think the complaint is mostly that if you enter a folder it's less self-evident how to paste in it.

**Comment by cramoisi** [comment]
@worldexplorer for 2. When you cut a epub and you want to paste it in an empty folder, you just have to long press on the folder itself (not inside the folder) and then choose "paste". It will do it right.

It works for all folder, you don't have to open them to paste something inside.

**Comment by BEEDELLROKEJULIANLOCKHART** [comment]
These suggestions and problems should be separated, because some I disagree with, some I arew with, and some I want to subscribe to whereas some I do not, however that is not possible with such an epic.

**Comment by Frenzie** [comment]
Nearly all of these have been implemented since this issue was opened.
